src/libcephsqlite.cc

```diff
@@ -353,7 +353,7 @@ static int Read(sqlite3_file *file, void *buf, int len, sqlite_int64 off)
     auto end = ceph::coarse_mono_clock::now();
     getdata(f->vfs).logger->tinc(P_OPF_READ, end-start);
     if (rc < len) {
-      memset(buf, 0, len-rc);
+      memset((unsigned char*)buf+rc, 0, len-rc);
```

Comment:
change looks correct for what it does 👍 just a couple followup questions:
1. when we return `SQLITE_IOERR_SHORT_READ`, how does sqlite figure out how many bytes are available?
2. why is it necessary to zero memory at all?

Reply:
> change looks correct for what it does 👍 just a couple followup questions:
> 
>     1. when we return `SQLITE_IOERR_SHORT_READ`, how does sqlite figure out how many bytes are available?

:magic_wand: 

>     2. why is it necessary to zero memory at all?

Because they said so: https://tracker.ceph.com/issues/62492 (see copy paste)

Reply:
sorry, i see that 2. was answered in the tracker issue: it's a requirement from the sqlite API